client/state/sites/reducer.js

```diff
@@ -103,8 +103,6 @@ export function items( state = {}, action ) {
 			}, initialNextState );
```

Comment:
There is a comment here that suggests us to remove the `if ( SITES_UPDATE === action.type && ! memo[ site.ID ] ) {` check when `sites-list` is removed. However, that can damage our use case of this action, right? We might want to remove that comment just in case, cc @aduth for a second opinion on that.